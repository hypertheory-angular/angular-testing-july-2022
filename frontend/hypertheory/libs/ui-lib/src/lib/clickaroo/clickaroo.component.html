<div class="alert alert-success">
  <p>This is a The Magic Button! Click it for a good time!</p>
  <input type="text" #msg />
  <button class="btn btn-primary" (click)="doIt(msg)">Click Me</button>
</div>
