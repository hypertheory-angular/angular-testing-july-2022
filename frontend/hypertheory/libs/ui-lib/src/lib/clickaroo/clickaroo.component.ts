import { Component, EventEmitter, OnInit, Output } from '@angular/core';

@Component({
  selector: 'hypertheory-clickaroo',
  templateUrl: './clickaroo.component.html',
  styleUrls: ['./clickaroo.component.css'],
})
export class ClickarooComponent {
  @Output() buttonClicked = new EventEmitter<string>();
  doIt(el: HTMLInputElement) {
    this.buttonClicked.emit(el.value);
    el.value = '';
    el.focus();
    // clear out the text box.
    // put the cursor focus back into it.
  }
}
