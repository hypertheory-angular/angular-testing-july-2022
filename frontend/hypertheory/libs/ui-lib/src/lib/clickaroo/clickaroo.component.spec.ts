import { Spectator, createComponentFactory } from '@ngneat/spectator/jest';

import { ClickarooComponent } from './clickaroo.component';
import '@ngneat/spectator/jest';
import { tap } from 'rxjs';
describe('ClickarooComponent', () => {
  let spectator: Spectator<ClickarooComponent>;
  const createComponent = createComponentFactory(ClickarooComponent);

  it('should create', () => {
    spectator = createComponent();

    expect(spectator.component).toBeTruthy();
  });
  it('clears out and focuses the input on button click', () => {
    spectator = createComponent();
    spectator.typeInElement('Hello, World', 'input[type="text"]');
    spectator.click('button');

    expect('input[type="text"]').toHaveValue('');
    expect('input[type="text"]').toBeFocused();
  });
  it('sends an output to the parent', () => {
    spectator = createComponent({ detectChanges: false });
    let result: string | undefined = undefined;
    spectator
      .output<string>('buttonClicked')
      .pipe(tap((value) => (result = value)))
      .subscribe();

    spectator.typeInElement('Hello, Tacos', 'input[type="text"]');
    spectator.click('button');

    spectator.detectChanges();
    expect(result).toBe('Hello, Tacos');
  });
  it('matches the snapshot', () => {
    spectator = createComponent();
    expect(spectator.fixture).toMatchSnapshot();
  });
});
