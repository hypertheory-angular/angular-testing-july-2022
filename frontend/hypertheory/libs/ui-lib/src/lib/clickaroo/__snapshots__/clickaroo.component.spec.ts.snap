// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickarooComponent matches the snapshot 1`] = `
<hypertheory-clickaroo
  __ngContext__={[Function Number]}
  buttonClicked={[Function EventEmitter_]}
>
  <div
    class="alert alert-success"
  >
    <p>
      This is a The Magic Button! Click it for a good time!
    </p>
    <input
      type="text"
    />
    <button
      class="btn btn-primary"
    >
      Click Me
    </button>
  </div>
</hypertheory-clickaroo>
`;
